client/dashboard/sites/scan/components/threat-description.tsx

```diff
@@ -0,0 +1,198 @@
+import { __, sprintf } from '@wordpress/i18n';
+import MarkedLines from 'calypso/components/marked-lines';
+import { CODEABLE_JETPACK_SCAN_URL } from '../constants';
+
+export function ThreatDescription( { threat }: { threat: Threat } ) {
+	const renderFixTitle = () => {
+		switch ( threat.status ) {
+			case 'fixed':
+				return __( 'How did Jetpack fix it?' );
+			case 'current':
+				if ( threat.fixable && threat.status === 'current' ) {
+					return __( 'How will we fix it?' );
+				}
+				return __( 'How to resolve or handle this detection?' );
+			default:
+				return __( 'How will we fix it?' );
+		}
+	};
+
+	const renderFilename = () => {
+		const { filename } = threat;
+		if ( ! filename ) {
+			return null;
+		}
+
+		return (
+			<>
+				<Text>
+					{
+						/* translators: filename follows in separate line; e.g. "PHP.Injection.5 in: `post.php`" */
+						__( 'Threat found in file:' )
+					}
+				</Text>
+				<Text as="pre">{ filename }</Text>
+			</>
+		);
+	};
+
+	const renderDatabaseRows = () => {
+		const { table, details, pk_column, value } = threat;
+
+		if ( ! table || ! details ) {
+			return null;
+		}
+
+		const row = {
+			table,
+			primary_key_column: pk_column,
+			primary_key_value: value,
+			details,
+		};
+		const content = JSON.stringify( row, null, '\t' ) + '\n';
+
+		return (
+			<>
+				<Text>
+					{ sprintf(
+						/* translators: %s: table name */
+						__( 'Threat found in the table %s, in the following rows:' ),
+						table
+					) }
+				</Text>
+				<Text as="pre">{ content }</Text>
+			</>
+		);
+	};
+
+	const getThreatFix = () => {
+		const { fixable } = threat;
+
+		if ( ! fixable ) {
+			return null;
+		}
+
+		switch ( fixable.fixer ) {
+			case 'replace':
+				return __( 'Jetpack Scan will replace the affected file or directory.' );
+			case 'delete':
+				return __( 'Jetpack Scan will delete the affected file or directory.' );
+			case 'update':
+				if ( fixable.target ) {
+					return sprintf(
+						/** translators: %s: version */
+						__( 'Jetpack Scan will update to a newer version (%s).' ),
+						fixable.target
+					);
+				}
+				return __( 'Jetpack Scan will update to a newer version.' );
+			case 'edit':
+				return __( 'Jetpack Scan will edit the affected file or directory.' );
+			case 'rollback':
+				if ( fixable.target ) {
+					return sprintf(
+						/** translators: %s: version */
+						__( 'Jetpack Scan will rollback the affected file to the version from %s.' ),
+						fixable.target
+					);
+				}
+				return __( 'Jetpack Scan will rollback the affected file to an older (clean) version.' );
+			default:
+				return __( 'Jetpack Scan will resolve the threat.' );
+		}
+	};
+
+	const renderFix = () => {
+		if ( threat.status === 'fixed' ) {
+			return;
+		}
+
+		if ( ! threat.fixable ) {
+			return (
+				<>
+					{ ! threat.rows && (
+						<Text>
+							{ __(
+								'Jetpack Scan cannot automatically fix this threat. We suggest that you resolve the threat manually: ' +
```

Comment:
Is it okay for the translation to use `+` 🤔

Reply:
Thanks for asking! I'm actually not sure :D, so I just removed the concatenation for now. I will ask later, because I'm curious too :)